### PR TITLE
Exclude directories when extracting module wordings

### LINKED ISSUE #24987: Translations file scanning should not go to vendor folder
#### Describe the bug

If a module has a vendor folder with bit more stuff the translation view is really slow and might either timeout or run out of memory.

The translation scanning in a module should not look into `vendor` folder (and probably `lib` and some others too).

#### Expected behavior

Vendor folder should not be scanned

#### Steps to Reproduce

Create minimal module with 

```
    public function isUsingNewTranslationSystem()
    {
        return true;
    }
```

Go to bo module translation for that module
=> the translation loads nicely

Now add a composer.json with some depencies (tested with `iio/libmergepdf`) and run

```
composer install --no-dev
```

go back to the module translation page
=> the page will either be really slow, timeout or run out of memory

#### Additional information

Some slack discussion about this:
image

* PrestaShop version: develop branch
* PHP version: (whatever comes from docker-compose up in develop branch right now :)

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Excludes `vendor`, `lib`, `tests` directories when scanning modules' directories to extract translation catalogue.<br>Update `prestashop/translationtools-bundle` to 4.1.0 and php-cs-fixer to 2.19.3
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #24987
| How to test?      | Use a module which has translations in vendor, lib or tests directory and check that these translations doesn't appear on translations interface.<br>For example, the module https://github.com/sowbiba/translationtesting has dependencies with https://github.com/sowbiba/translationtesting-dependency which has translated strings (https://github.com/sowbiba/translationtesting-dependency/blob/master/views/templates/admin/configure.tpl#L30). Those translated messages may not appear in the BO page<br> The performances problems described in the issue may not occurred anymore
| Possible impacts? | -

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24990)

## Breaking Changes

 - LegacyModuleExtractor now requires an extra parameter catalogueExtractExcludedDirectories to define folders to exclude
 - In LegacyModuleExtractor we use specific class for phpExtractor, smartyExtractor and twigExtractor instead of ExtractorInterface, because the excludedDirs feature is in the TraitExtractor and not defined in the Interface

### CODE REVIEW

File: src/Core/Translation/Storage/Extractor/LegacyModuleExtractor.php [hunk 1]
```diff
@@ -59,21 +62,29 @@ final class LegacyModuleExtractor implements LegacyModuleExtractorInterface
...
     public function __construct(
-        ExtractorInterface $phpExtractor,
-        ExtractorInterface $smartyExtractor,
-        ExtractorInterface $twigExtractor,
-        string $modulesDirectory
+        AbstractFileExtractor $phpExtractor,
+        AbstractFileExtractor $smartyExtractor,
+        AbstractFileExtractor $twigExtractor,
+        string $modulesDirectory,
+        array $catalogueExtractExcludedDirectories
```
Comment (matks) [inline src/Core/Translation/Storage/Extractor/LegacyModuleExtractor.php:80]: BC break

**Review by matks — CHANGES_REQUESTED** [review]
PR OK but please add BC break description in PR body

Verify AbstractFileExtractor usage

**Review by matks — DISMISSED** [review]

File: classes/assets/AbstractAssetManager.php [hunk 2]
```diff
@@ -28,6 +28,8 @@
 
 abstract class AbstractAssetManagerCore
 {
+    use PrestaShop\PrestaShop\Adapter\Assets\AssetUrlGeneratorTrait;
```
Comment (eternoendless) [inline classes/assets/AbstractAssetManager.php]: IMO these unrelated changes should be best left out of the PR

Comment (sowbiba) [reply classes/assets/AbstractAssetManager.php]: I updated php-cs-fixer that's why these changes are here :(

**Comment by matks** [comment]
Re-running CI

**Review by matks — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by sowbiba** [comment]
/home/isow/Téléchargements/translationtesting.zip

**Review by jolelievre — APPROVED** [review]
Thanks @sowbiba

**Review by matthieu-rolland — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @sowbiba,

It is ok :heavy_check_mark: 
I checked with a test module provided by @sowbiba 
https://watch.screencastify.com/v/X6V93hzmvd036tVcsQpK
I checked the test module https://github.com/tswfi/translationtesting
https://watch.screencastify.com/v/l05DiLbuyz8DRioFU5AQ

Thanks!

**Pull request merged**